docs/developer/architecture/control-plane-ingress.rst

```diff
@@ -0,0 +1,210 @@
+Available, so if you lose one node, including Bootstrap Node, you can still
+access various UIs.
+NOTE: In this document, we do not talk about the High Availability of the
+components itself but really only to the access through the Ingress
+(e.g.: We do not want to solve salt-master HA here).
+
+User Stories
+------------
+
+MetalK8s and Grafana UIs HA
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+I have a multi-node MetalK8s cluster, with at least 3 Control Plane nodes
+if I lose one of the Control Plane nodes (including the Bootstrap one)
+I can still access and authenticate on the MetalK8s and Grafana UIs.
+
+Design Choices
+--------------
+
+To have a proper HA Control Plane Ingress we want to use a
+`Virtual IP <https://en.wikipedia.org/wiki/Virtual_IP_address>`_ using
+`MetalLB <https://metallb.universe.tf/>`_ so that we can rely on layer2
+ARP requests when possible or let MetalLB advertise IPs using BGP when
+the user router support it.
+
+NOTE: In this document we talk about MetalLB used for Control Plane
+Ingress and not for Workload Plane.
+
+But in some network, it may be not possible to use one of those methods
+so we also let the possibility to not use MetalLB but instead just
+assign an External IP, provided by the user, that we expect to be a
+Virtual IP, and we do not manage on our side but it’s managed by the user
+using whatever mechanism to switch this IP between Control Plane nodes.
+
+To summarize 3 different deployments possible depending on the user
+environment:
+
+- Using VIP managed by Layer2 ARP with MetalLB (builtin MetalK8s)
+- Using VIP managed by BGP announce with MetalLB (builtin MetalK8s)
+- Using a user-provided IP that should switch between Control
+  Plane nodes (managed by the user)
+
+NOTE: In all those 3 approaches we want the user to provide the Control
+Plane Ingress IP he wants to use.
+
+Rejected Design Choices
+-----------------------
+
+Manage Virtual IP by MetalK8s
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of using MetalLB to manage the Virtual IP we could have manage this
+Virtual IP directly in MetalK8s with KeepAliveD (or any other “HA tool”) this
+way we really controle the Virtual IP.
+
+This approach was rejected because it seems to do not provide any real
+advantage compare to use MetalLB directly that will manage this Virtual IP
+for us and may provide a bunch of other useful feature for the future.
+
+Rely on DNS resolution
+~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of using a Virtual IP we could rely on DNS resolution use this
+FQDN to configure Control Plane components. In this case we let the user
+configure his DNS to resolve on an IP of one Control Plane node.
+
+WIth this approach we let the DNS server handle the “High Availability” of
+the Control Plane UIs, basically if we lose the node that resolve the DNS
+then we expect the DNS server to switch to another IP of a working Control
+Plane node.
+
+This approach was rejected because it’s not a real HA and we expect DNS server
```

Comment:
It may be transparent if the DNS server is active and properly switch from an IP to another in case of failure, but as is it's a "False HA" since we do not have any HA we just move the "problem": "we do not need the IP to be HA but a specific DNS"